obal/data/roles/fetch_sources/tasks/main.yml

```diff
@@ -112,22 +117,12 @@
         - '%global _release'
         - '%global release'
 
-    - name: 'add changelog entry'
-      lineinfile:
-        path: "{{ spec_file_path }}"
-        line: |
-          * {{ lookup('pipe', 'date +"%a %b %d %Y"') }} {{ downstream_changelog_name | default('Satellite6 Jenkins') }} {{ downstream_version }}-{{ downstream_release }}
-          {{ downstream_changelog }}
-        insertafter: '^%changelog'
+    - name: 'ensure changelog'
+      changelog:
+        spec: "{{ spec_file_path }}"
+        changelog: "{{ changelog }}"
+      when: changelog is defined
```

Comment:
`changelog` is always defined, as we have a `set_fact` above, so maybe this `when` is not needed?

Reply:
Double check the `when` conditions on the set_fact above. This is not always defined. For example, when copying a package from an upstream git repo to a downstream git repo. Our tests are what caught this little nugget.

Reply:
ah, right!